microprofile/security/src/main/java/io/helidon/microprofile/security/SecurityContextProvider.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright (c) 2019 Oracle and/or its affiliates. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.helidon.microprofile.security;
+
+import java.util.concurrent.atomic.AtomicReference;
+
+import javax.enterprise.inject.Produces;
+
+import io.helidon.security.SecurityContext;
+
+/**
+ * A bean for injection of security context.
+ */
+class SecurityContextProvider {
+    private final AtomicReference<SecurityContext> currentReference = new AtomicReference<>();
+
+    SecurityContextProvider() {
+    }
+
+    @Produces
+    SecurityContext securityContext() {
+        return currentReference.get();
+    }
+
+    void securityContext(SecurityContext toSet) {
+        this.currentReference.set(toSet);
```

Comment:
Do I understand right that the only class that ever calls this method is a JAX-RS filter?

Reply:
yes

Reply:
Removed.